app/src/main/java/app/revanced/integrations/settingsmenu/ReVancedSettingsFragment.java

```diff
@@ -400,6 +408,51 @@ private void setPatchesInformation() {
             integrations.setSummary(BuildConfig.VERSION_NAME);
     }
 
+    /**
+     * Add Preference to Whitelist settings submenu
+     */
+    public void AddWhitelistSettings() {
+        try {
+            Activity activity = ReVancedSettingsFragment.this.getActivity();
+            boolean isIncludedSB = PatchStatus.SponsorBlock();
+            boolean isIncludedSPEED = PatchStatus.VideoSpeed();
+            boolean isIncludedADS = PatchStatus.VideoAds();
+
+            if (isIncludedSB || isIncludedSPEED || isIncludedADS) {
+                // Sponsorblock
+                if (isIncludedSB) {
+                    Whitelist.setEnabled(WhitelistType.SPONSORBLOCK, SettingsEnum.SB_WHITELIST.getBoolean());
+
+                    WhitelistedChannelsPreference WhitelistSB = new WhitelistedChannelsPreference(activity);
+                    WhitelistSB.setTitle(str("revanced_whitelisting_sponsorblock"));
+                    WhitelistSB.setWhitelistType(WhitelistType.SPONSORBLOCK);
+                    this.whitelistingPreferenceScreen.addPreference(WhitelistSB);
+                }
+
+                // Video Speed
+                if (isIncludedSPEED) {
+                    Whitelist.setEnabled(WhitelistType.SPEED, SettingsEnum.SPEED_WHITELIST.getBoolean());
+
+                    WhitelistedChannelsPreference WhitelistSPEED = new WhitelistedChannelsPreference(activity);
+                    WhitelistSPEED.setTitle(str("revanced_whitelisting_speed"));
+                    WhitelistSPEED.setWhitelistType(WhitelistType.SPEED);
+                    this.whitelistingPreferenceScreen.addPreference(WhitelistSPEED);
+                }
+            } else {
+                //True is disable
```

Comment:
@inotia00 I am using `enableDisablePreferences` but look like it is an incorrect code for this case. Here is the old code:
```java
SwitchPreference setWhitelist = (SwitchPreference) findPreferenceOnScreen(SettingsEnum.OVERLAY_BUTTON_WHITELIST.getPath());
this.overlayPreferenceScreen.removePreference(setWhitelist);
this.overlayPreferenceScreen.removePreference(whitelistingPreferenceScreen);